default.nix

```
@@ -195,7 +199,8 @@ let
               while IFS= read -r c
               do
                 echo "Running npm command: $c"
-                $c || (echo "$c: failure, aborting" && kill $napalm_REGISTRY_PID && exit 1)
+                ip2unix -r out,addr=127.0.0.100,path="$registrySock" -- $c \
+                  || (echo "$c: failure, aborting" && kill $napalm_REGISTRY_PID && exit 1)
```

```suggestion
                  || (echo "$c: failure, aborting" && (rm -rf $(dirname $registrySock) || true) && kill $napalm_REGISTRY_PID && exit 1)
```

---

The socket file should already be cleaned up by `ip2unix`, but since the kill signal is `SIGTERM` it depends on the signal handler of the registry. Nevertheless, if we assume that things are running unsandboxed *and* are not cleaned up properly by Nix, I'd probably avoid unquoted `rm -rf` at all costs, eg. more like this (without the extra directory, as mentioned above):

```suggestion
                  || (echo "$c: failure, aborting" && kill "$napalm_REGISTRY_PID" && rm -f "$TMPDIR/registry.sock"; exit 1)
```